metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -68,6 +71,37 @@
 # patch the implementation.
 
 
+def bigquery_audit_metadata_query_template(dataset: str) -> str:
+    """
+    Receives a dataset (with project specified) and returns a query template that is used to query exported
+    AuditLogs containing protoPayloads of type BigQueryAuditMetadata.
+    :param dataset: the dataset to query against in the form of $PROJECT.$DATASET
+    :return: a query template, when supplied start_time and end_time, can be used to query audit logs from BigQuery
+    """
+    query = (
+        f"""
+    SELECT
+        timestamp,
+        logName,
+        insertId,
+        protopayload_auditlog AS protoPayload,
+        protopayload_auditlog.metadataJson AS metadata
+    FROM
+        `{dataset}.cloudaudit_googleapis_com_data_access`
```

Comment:
Should we plan to support log tables being partitioned? Maybe take in a flag and append the datepartition if the flag is true.

Right now, by default when setting up the logging sink, GCP created a time partitioned table. And so the ingestion  source failed with table not found error.

Reply:
Oh I see! Yeah, it'll make sense to support date-sharded tables. Let me update the PR.